kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/JobServiceMigrationIntegrationTest.java

```diff
@@ -0,0 +1,99 @@
+import org.kie.server.api.model.instance.RequestInfoInstance;
+import org.kie.server.integrationtests.shared.KieServerDeployer;
+import org.kie.server.integrationtests.shared.KieServerSynchronization;
+
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.kie.internal.executor.api.STATUS.DONE;
+import static org.kie.internal.executor.api.STATUS.ERROR;
+
+public class JobServiceMigrationIntegrationTest extends JbpmKieServerBaseIntegrationTest {
+
+    private static final String PROCESS_ID = "wih.process";
+    private static final String KJARS_SOURCES_PATH = "/kjars-sources/";
+    private static final String CONTAINER_ID1 = "async-exec-project";
+    private static final String CONTAINER_ID2 = "async-exec-project-101";
+
+    private static ReleaseId releaseId = new ReleaseId("org.kie.server.testing", "async-exec-project",
+            "1.0.0.Final");
+    
+    private static ReleaseId releaseId101 = new ReleaseId("org.kie.server.testing", "async-exec-project",
+            "1.0.1.Final");
+
+    
+    @BeforeClass
+    public static void buildAndDeployArtifacts() {
+        KieServerDeployer.buildAndDeployCommonMavenParent();
+        KieServerDeployer.buildAndDeployMavenProjectFromResource(KJARS_SOURCES_PATH+CONTAINER_ID1);
+        kieContainer = KieServices.Factory.get().newKieContainer(releaseId);
+        
+        KieServerDeployer.buildAndDeployMavenProjectFromResource(KJARS_SOURCES_PATH+CONTAINER_ID2);
+        kieContainer = KieServices.Factory.get().newKieContainer(releaseId101);
+
+        createContainer(CONTAINER_ID1, releaseId);
+        createContainer(CONTAINER_ID2, releaseId101);
+    }
+    
+    @AfterClass
+    public static void disposeContainers() {
+        disposeAllContainers();
+    }
+    
+    @Test
+    public void testMigrateAndRequeueFailingJob() throws Exception {
+        Long pid = processClient.startProcess(CONTAINER_ID1, PROCESS_ID);
+        
+        // short wait before querying the async job id
+        Thread.sleep(2000);
+        
+        // first container has a wrong deployment-structure (without CustomWorkItemHandler definition), so job ends up in Error status
+        List<RequestInfoInstance> errors = jobServicesClient.getRequestsByContainer(CONTAINER_ID1, singletonList(ERROR.toString()), 0, 10);
+        assertTrue(errors.size() > 0);
+        
+        Long jobId = errors.get(0).getId();
+        
+        // migrate process instance to container 2 (right deployment-structure containing the CustomWorkItemHandler definition)
+        MigrationReportInstance report = processAdminClient.migrateProcessInstance(CONTAINER_ID1, pid, CONTAINER_ID2, PROCESS_ID);
+        assertNotNull(report);
+        assertTrue(report.isSuccessful());
+        
+        jobServicesClient.updateRequestData(jobId, null, new HashMap<>());
```

Comment:
why do we need to update the request data? is it needed?

Reply:
Not really needed, removed, thanks for realizing @afalhambra